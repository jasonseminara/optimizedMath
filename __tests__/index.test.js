const { optimizedMath, getMask } = require('../');

describe('optimizedMath', () => {
  describe('#getMask', () => {
    it('should return 3 for a number that is divisible by 2 AND 3 [12]', () => {
      const expected = 3;
      expect(getMask(12)).toBe(expected);
    });

    it('should return 3 for a number that is divisible by 2 AND 3 [24]', () => {
      const expected = 3;
      expect(getMask(24)).toBe(expected);
    });

    it('should return 3 for a number that is divisible by 2 AND 3 [90]', () => {
      const expected = 3;
      expect(getMask(90)).toBe(expected);
    });

    it('should return 2 for a number that is divisible 3 [99]', () => {
      const expected = 2;
      expect(getMask(99)).toBe(expected);
    });

    it('should return the right code for each number', () => {
      const actual = Array(200).fill(0).map((n, i) => {
        if (i === 0) return undefined;
        return getMask(i);
      });
      expect(actual).toMatchSnapshot();
    });
  });

  describe('#optimizedMath', () => {
    console.log = jest.fn();
    it('should match the whole run and output', () => {
      optimizedMath(100);
      const actual = console.log.mock.calls.map(([call]) => call);
      expect(actual).toMatchSnapshot();
    });
  });
});
