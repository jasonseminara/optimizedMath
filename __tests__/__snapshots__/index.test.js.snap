// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`optimizedMath #getMask should return the right code for each number 1`] = `
Array [
  undefined,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
  1,
  2,
  1,
  0,
  3,
  0,
]
`;

exports[`optimizedMath #optimizedMath should match the whole run 1`] = `
Array [
  "The number '1' is odd.",
  "The number '2' is even.",
  "The number '3' is divisible by three.",
  "The number '4' is even.",
  "The number '5' is odd.",
  "The number '6' is divisible by two and three.",
  "The number '7' is odd.",
  "The number '8' is even.",
  "The number '9' is divisible by three.",
  "The number '10' is even.",
  "The number '11' is odd.",
  "The number '12' is divisible by two and three.",
  "The number '13' is odd.",
  "The number '14' is even.",
  "The number '15' is divisible by three.",
  "The number '16' is even.",
  "The number '17' is odd.",
  "The number '18' is divisible by two and three.",
  "The number '19' is odd.",
  "The number '20' is even.",
  "The number '21' is divisible by three.",
  "The number '22' is even.",
  "The number '23' is odd.",
  "The number '24' is divisible by two and three.",
  "The number '25' is odd.",
  "The number '26' is even.",
  "The number '27' is divisible by three.",
  "The number '28' is even.",
  "The number '29' is odd.",
  "The number '30' is divisible by two and three.",
  "The number '31' is odd.",
  "The number '32' is even.",
  "The number '33' is divisible by three.",
  "The number '34' is even.",
  "The number '35' is odd.",
  "The number '36' is divisible by two and three.",
  "The number '37' is odd.",
  "The number '38' is even.",
  "The number '39' is divisible by three.",
  "The number '40' is even.",
  "The number '41' is odd.",
  "The number '42' is divisible by two and three.",
  "The number '43' is odd.",
  "The number '44' is even.",
  "The number '45' is divisible by three.",
  "The number '46' is even.",
  "The number '47' is odd.",
  "The number '48' is divisible by two and three.",
  "The number '49' is odd.",
  "The number '50' is even.",
  "The number '51' is divisible by three.",
  "The number '52' is even.",
  "The number '53' is odd.",
  "The number '54' is divisible by two and three.",
  "The number '55' is odd.",
  "The number '56' is even.",
  "The number '57' is divisible by three.",
  "The number '58' is even.",
  "The number '59' is odd.",
  "The number '60' is divisible by two and three.",
  "The number '61' is odd.",
  "The number '62' is even.",
  "The number '63' is divisible by three.",
  "The number '64' is even.",
  "The number '65' is odd.",
  "The number '66' is divisible by two and three.",
  "The number '67' is odd.",
  "The number '68' is even.",
  "The number '69' is divisible by three.",
  "The number '70' is even.",
  "The number '71' is odd.",
  "The number '72' is divisible by two and three.",
  "The number '73' is odd.",
  "The number '74' is even.",
  "The number '75' is divisible by three.",
  "The number '76' is even.",
  "The number '77' is odd.",
  "The number '78' is divisible by two and three.",
  "The number '79' is odd.",
  "The number '80' is even.",
  "The number '81' is divisible by three.",
  "The number '82' is even.",
  "The number '83' is odd.",
  "The number '84' is divisible by two and three.",
  "The number '85' is odd.",
  "The number '86' is even.",
  "The number '87' is divisible by three.",
  "The number '88' is even.",
  "The number '89' is odd.",
  "The number '90' is divisible by two and three.",
  "The number '91' is odd.",
  "The number '92' is even.",
  "The number '93' is divisible by three.",
  "The number '94' is even.",
  "The number '95' is odd.",
  "The number '96' is divisible by two and three.",
  "The number '97' is odd.",
  "The number '98' is even.",
  "The number '99' is divisible by three.",
  "The number '100' is even.",
]
`;
